zorder/src/main/scala/org/locationtech/sfcurve/zorder/Z2.scala

```diff
@@ -31,12 +32,11 @@ class Z2(val z: Long) extends AnyVal {
   def d1 = dim(1)
 
   def mid(p: Z2): Z2 = {
-    var ans: Z2  = new Z2(0)
-    if (p.z < z)
-      ans  = new Z2(p.z + (z - p.z)/2)
-    else
-      ans = new Z2(z + (p.z - z)/2)
-    ans
+    if (p.z < z) {
```

Comment:
This may not apply, but some of the arithmetic gets weird from this expression when the values approach ```Long.MinValue``` and ```Long.MaxValue```... for example:  ```Z2(Long.MaxValue).mid(Long.MinValue) = Long.MaxValue```, which I suspect is not intended.  As a variant, consider:

```scala
def mid(p: Z2): Z2 = {
  new Z2((p.z >> 1) + (z >> 1) + ((p.z & 2 + z & 2) >> 2))
}
```

That yields ```Z2(Long.MaxValue).mid(Long.MinValue) = -1```.  If Z2 values are never negative, then perhaps this isn't an issue at all (although we do eliminate a conditional expression at the same time).  Anyway, just wonderin'.  Thanks!